src/ReinforcementLearningFarm/src/algorithms/explorers/epsilon_speedy_explorer.jl

```diff
@@ -0,0 +1,92 @@
+
+using ReinforcementLearningCore
+using ReinforcementLearningBase
+import ReinforcementLearningBase: RLBase
+
+using FillArrays: Trues
+using Random
+using Distributions: Categorical
+using Base
+
+"""
+    EpsilonSpeedyExplorer(β::Float64)
+
+`EpsilonSpeedyExplorer` is an explorer that selects the action with the maximum value with probability `1 - ϵ` and selects a random action with probability `ϵ`.
+The probability of selecting a random action is given by `exp(β * -t)`, where `t` is the number of times `plan!` has been called.
+`EpsilonSpeedyExplorer` differs from `EpsilonGreedyExplorer` in that it uses the `exp` function to calculate the probability of selecting a random action over the full range of `t` and only accepts one argument, `β`.
+
+"""
+struct EpsilonSpeedyExplorer{R} <: AbstractExplorer
+    β::Float64
+    β_neg::Float64
+    step::Base.RefValue{Int}
```

Comment:
Is it really faster to use a Ref instead of a mutable struct with ::Int ?

Reply:
good question. timings are identical:

```
# 21.079 ns (<0.01 allocs: 0.130 bytes)
# 22.028 ns (<0.01 allocs: 0.136 bytes)
```

When there's only one struct element that needs to be mutable and it's a float, I like to use a `Ref` to eliminate accidental mutations, happy to change this...